### PR TITLE
Updated black formatter version in `.pre-commit-config.yaml` file and modified all the files according to it.

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
# PR Description 
[New version of black](https://github.com/psf/black/releases) is released, So I have updated the version in the `.pre-commit-config.yaml` file.
Also, in this release they mentioned the following things:
image

So, I have updated the `.pre-commit-config.yaml` file accordingly by changing the link to this (https://github.com/psf/black-pre-commit-mirror), and modified all the files according to the latest version of black by running `pre-commit run --all-files`.

## Checklist 
- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### Socials: 
https://twitter.com/Sai_Suraj_27

### CODE REVIEW

**Comment by Sai-Suraj-27** [comment]
Already few big organizations like [django](https://github.com/django/django/blob/ecb6085f6fdb6319656b54cc92e6a91c1df47f23/.pre-commit-config.yaml#L4), [poetry](https://github.com/django/django/blob/ecb6085f6fdb6319656b54cc92e6a91c1df47f23/.pre-commit-config.yaml#L4), [mypy](https://github.com/python/mypy/blob/ed18fea5b17ef3a969b37b4906dd7c237ddb1825/.pre-commit-config.yaml#L8)...etc have updated their `.pre-commit-config.yaml` file accordingly by changing the link to this (https://github.com/psf/black-pre-commit-mirror).
@KareemMAX Can you please review this too, Thank you.

**Review by vedpatwardhan — COMMENTED** [review]
Looks good on a high-level, but I'll leave it with @KareemMAX to provide more feedback. Thanks @Sai-Suraj-27 😄

**Review by KareemMAX — CHANGES_REQUESTED** [review]
Hi @Sai-Suraj-27,
To be sure that everything is working as planned, can you remove all the changes `black` made and only leave `.pre-commit-config.yml`? You may use `--no-verify` while committing. As it'll simplify the review process to a simple file.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -5,13 +5,10 @@ repos:
...
       - id: check-toml
-  - repo: https://github.com/psf/black
-    rev: 23.7.0
+  - repo: https://github.com/psf/black-pre-commit-mirror
+    rev: 23.9.1
     hooks:
       - id: black
-        language_version: python3
-        args:
-          - "--preview"
```
Comment (KareemMAX) [inline .pre-commit-config.yaml:14]: Why did you remove those?

**Review by KareemMAX — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (Sai-Suraj-27) [reply .pre-commit-config.yaml]: @KareemMAX Sorry, I removed by mistake. Now I have removed the changes made by black as you said and only updated the ` .pre-commit-config.yml` file, Thank you.

**Review by KareemMAX — APPROVED** [review]
